### PR TITLE
Do not reset text selection when the TextBox loses focus

### LINKED ISSUE #17151: TextBox should not clear selection on focus loss
### Is your feature request related to a problem? Please describe.

Right now, when a TextBox loses focus, it clears the selection.  This prevents external controls, like toolbar buttons, from obtaining the TextBox selection and interacting with it since it is now gone.  Further, when re-Tabbing back into the control, the selection is not restored.

### Describe the solution you'd like

Instead of clearing the selection, the TextBox should simply not render the selected text differently (as "selected") when focus is outside of the control.  This allows external controls to access the selection, and keeps things visually the same as before.  And then when focus moves back into the control, the selection is restored and redisplayed unless otherwise programmatically altered.

One thing to be careful of is when the TextBox's context menu/flyout is open, you still want to render the selected text as "selected".

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

This PR removes some code that explicitly resets the current selection when the TextBox loses focus. We also no longer render the text selection when the TextBox has no focus to match what other frameworks do.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Fixed issues

Fixes: #17151

### CODE REVIEW

**Review by MrJul — APPROVED** [review]
LGTM!
Do we want to introduce an option to keep the old behavior?

**Comment by Gillibald** [comment]
We could add such option and change the default in a future release y

**Comment by rabbitism** [comment]
https://github.com/AvaloniaUI/Avalonia/issues/11693 can also be closed.

**Comment by Gillibald** [comment]
> #11693 and #16825 can also be closed.

Okay I will implement #11693 that looks to be a good change

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 1]
```diff
@@ -17,6 +17,9 @@ namespace Avalonia.Controls.Presenters
 {
     public class TextPresenter : Control
     {
+        public static readonly StyledProperty<bool> ShowSelectionHighlightProperty =
```
Comment (MrJul) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs:20]: Shouldn't this property be added to the switch in `OnPropertyChanged` that triggers the call to `InvalidTextLayout`?

**Pull request merged**